src/CaptureClient/include/CaptureClient/ApiEventIdProvider.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CAPTURE_CLIENT_API_EVENT_ID_PROVIDER_H_
+#define CAPTURE_CLIENT_API_EVENT_ID_PROVIDER_H_
+
+#include <cstdint>
+
+#include "ClientData/TimerTrackDataIdManager.h"
+#include "GrpcProtos/capture.pb.h"
+
+namespace orbit_capture_client {
+
+// The inferface defines a map from timer_info to ids.
+// When called twice of identical `timer_info` instances,
+// it should return the same ids.
+class ApiEventIdProvider {
+ public:
+  virtual ~ApiEventIdProvider() = default;
+
+  [[nodiscard]] virtual uint64_t ProvideId(const TimerInfo& timer_info) = 0;
+};
+
+// Sets equal `api_scope_group_id` to the instances of TimeInfo
+// if and only if their `api_scope_name` are equal.
+// The ids are chosen consequtively starting with `start_id`.
+//
+// To instantiate, use `NameEqualityApiEventIdSetter::Create`
```

Comment:
Can the constructors be private then?

Reply:
Either that, or also let the constructor take the capture_options.

Reply:
I am currently calling `std::make_unique<NameEqualityApiEventIdProvider>`in `Create`. This needs a public constructor. If constructors are private, I'd have to use `new` directly. Is that OK? 

I don't like passing capture_options to constructor, as constructor should be simple (the style guide prohibits only virtual method calls though).